Source/SteveSloth/Private/MyPlayer.cpp

```diff
@@ -232,6 +260,22 @@ void AMyPlayer::SetupPlayerInputComponent(UInputComponent* PlayerInputComponent)
 
 		inputComponent->BindAction(PAiming, ETriggerEvent::Triggered, this, &AMyPlayer::Aiming);
 		inputComponent->BindAction(PAiming, ETriggerEvent::Completed, this, &AMyPlayer::AimingStop);
+
+		inputComponent->BindAction(PGrappplingHook, ETriggerEvent::Triggered, this, &AMyPlayer::GrapplingHook);
+		inputComponent->BindAction(PGrappplingHook, ETriggerEvent::Completed, this, &AMyPlayer::GrapplingHook);
+
+		inputComponent->BindAction(PClimbingClaw, ETriggerEvent::Triggered, this, &AMyPlayer::ClimbingClaw);
+		inputComponent->BindAction(PClimbingClaw, ETriggerEvent::Completed, this, &AMyPlayer::ClimbingClaw);
+
+		inputComponent->BindAction(PShovel, ETriggerEvent::Triggered, this, &AMyPlayer::Shovel);
+		inputComponent->BindAction(PShovel, ETriggerEvent::Completed, this, &AMyPlayer::Shovel);
+
+		inputComponent->BindAction(PPropeller, ETriggerEvent::Triggered, this, &AMyPlayer::Propeller);
+		inputComponent->BindAction(PPropeller, ETriggerEvent::Completed, this, &AMyPlayer::Propeller);
+
+		inputComponent->BindAction(PMagnetic, ETriggerEvent::Triggered, this, &AMyPlayer::Magnetic);
+		inputComponent->BindAction(PMagnetic, ETriggerEvent::Completed, this, &AMyPlayer::Magnetic);
+
```

Comment:
we are never going to use all the functions at the same time. So the functions should not have their own input, should just be one input with a switch state inside it that calls the function based off of which upgrade is being used at that time.